File: Famous Algorithms/findingSubsequence.py
def subSequence(word):

	if(len(word) == 0):
		return ""

	first = word[0:1]

	rest = subSequence(word[1:])

	result = ""

	for subseq in rest.split(","):
		result+= "," + first + subseq
		result+= "," + subseq

	return result[1:]

output = subSequence("abc")
print(output)

File: Famous Algorithms/findingSubsequence.java
class Main{
	private static String getSubSequence(String word){

		if(word.isEmpty()){
			return "";
		}

		char first = word.charAt(0);

		String rest = getSubSequence(word.substring(1));
        
        String result = "";
		for(String subseq : rest.split(",", -1)){
			result+= "," + first + subseq;
			result+= "," + subseq;
		}

		return result.substring(1);
	}

	public static void main(String[] args){

		String word = "abc";

		String subsequence = getSubSequence(word);

		System.out.println(subsequence);

	}
}

//time complexity is O(2^N)
//space complexity is O(N)